CT-1657 use config state machine for non-trigger FOI in awaiting responder

stephenrichards opened the pull request.

misaka commented on public/nginx/505.html:
```diff
@@ -0,0 +1,101 @@
+        <meta charset="utf-8" />
+        <title>This website is unavailable at the moment - GOV.UK Contact</title>
+
+        <link rel="stylesheet" media="all" href="/assets/application.css" />
```
So the problem here is that `/assets` will go to the Rails app, won't it? We should probably switch `/assets` for `/static` in the 5XX files and send Platforms the various asset files we need, requesting they make them available there. If we just use the existing files we don't have to copy anything into a separate place in the repo or do any further processing ([like we did with the public tool](https://github.com/ministryofjustice/correspondence_tool_public/tree/develop/public/files)) ... we can just precompile assets, maybe run `rake assets:non_digested` (for convenience), and give the the files we need to Platforms.

EdwardAndress replied:
Done, I think!

stephenrichards committed:
CT-1657 Non-trigger FOI in awaiting responder use Configuratble State Machine

This PR changes the awaiting_responder state for non-trigger FOI cases to use
the configurable state machine.  It also introduces the concept of different
workflows for trigger and non trigger cases, which is recorded on the case and
on each transition.

There is a temporary test which tests the state, the state machine and transition
after every action in the end-to-end user journeys which we will use while
converting the other states to use the configuratble state machine.

misaka reviewed (changes requested):
I'm going to save reviewing the specs for once the changes mentioned are done ... just so I have a little bit of a break. :)

stephenrichards committed:
CT-1567 Remove setting new workflow in service - is set by state machine

stephenrichards committed:
CT-1567 Model Annotations

stephenrichards committed:
CT-1567 Rename migration back to original

stephenrichards committed:
CT-1567 Fix typp
